File: app/src/main/java/com/fastaccess/provider/timeline/handler/QouteHandler.java
package com.fastaccess.provider.timeline.handler;

import android.support.annotation.ColorInt;
import android.text.SpannableStringBuilder;

import com.zzhoujay.markdown.style.MarkDownQuoteSpan;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

import lombok.AllArgsConstructor;

/**
 * Created by Kosh on 23 Apr 2017, 11:30 AM
 */

@AllArgsConstructor public class QouteHandler extends TagNodeHandler {

    @ColorInt private int color;

    public void handleTagNode(TagNode node, SpannableStringBuilder builder, int start, int end) {
        builder.append("\n");
        builder.setSpan(new MarkDownQuoteSpan(color), start, builder.length(), 33);
        builder.append("\n");
    }

}

File: markdown/src/main/java/com/fastaccess/markdown/spans/QouteHandler.kt
package com.fastaccess.markdown.spans

import android.text.SpannableStringBuilder
import net.nightwhistler.htmlspanner.SpanStack

import net.nightwhistler.htmlspanner.TagNodeHandler

import org.htmlcleaner.TagNode

/**
 * Created by Kosh on 23 Apr 2017, 11:30 AM
 */

class QouteHandler(private val color: Int = 0) : TagNodeHandler() {
    override fun handleTagNode(node: TagNode?, builder: SpannableStringBuilder?, start: Int, end: Int, spanStack: SpanStack?) {
        builder?.let {
            builder.append("\n")
            builder.setSpan(MarkDownQuoteSpan(color), if (start > builder.length - 1) start + 1 else start, builder.length - 1, 33)
            builder.append("\n")
        }
    }
}
